pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -118,6 +131,54 @@
   private volatile long _lastIndexedTimeMs = Long.MIN_VALUE;
   private volatile long _latestIngestionTimeMs = Long.MIN_VALUE;
 
+  private static final ScheduledExecutorService _scheduledExecutorService;
+  private static final ConcurrentLinkedQueue<RealtimeLuceneReadersForRealtimeSegment> _luceneRealtimeReaders;
+  private static final ConcurrentHashMap<String, RealtimeLuceneReadersForRealtimeSegment> _segmentToRealtimeLuceneReadersMap;
+
+  static {
```

Comment:
This code should go into the segment data manager instead of a static here.

Reply:
I don't follow -- it has to be static right? We can't create an instance of this per LLRealtimeSegmentDataManager which also has a corresponding instance of MutableSegmentImpl.

Reply:
As discussed, it needs to be either an instance per table or a singleton instance across all tables. Best to create a class and add functionality to it. We can wire it in later depending on what else we decide (e.g. per table config to control the knob of consistency vs overhead)

Reply:
Done. Please look at RealtimeLuceneIndexRefreshState and RealtimeLuceneIndexReaderRefreshThread